### ISSUE TITLE
Are TCP/RTU adapters supported?

### ISSUE THREAD

**Issue by andig** [grid-x/modbus#20]
I'm trying to use a [Waveshare RS485](https://www.amazon.de/gp/product/B07S2K41MS/ref=ppx_yo_dt_b_asin_title_o02_s00?ie=UTF8&th=1) to Ethernet adapter, but fail to connect to the device (i/o timeout). Using this lib it seems I'm actually communicating like this:

        modbus TCP     modbus RTU
    client ----> adapter ---->device

Now I've seen https://github.com/goburrow/modbus/pull/42 and was wondering if that's how that works. Do these adapters typically convert modbus TCP to modbus RTU or do I need to "speak" RTU over TCP for that to work?

In other words: is the ref'ed PR relevant for this repo?

**Comment by frzifus** [comment]
i would guess that they are using modbus tcp and the device adds the checksum by itself. But even if not it should not be a big problem to transfer the checksum. Is there no online documentation for the device?

**Comment by andig** [comment]
> i would guess that they are using modbus tcp and the device adds the checksum by itself. 

Thought so. Might be that the device is just broken, I'll return it and try a different brand.

**Comment by frzifus** [comment]
i would suggest to monitor the input and output of the device. There you should see if the crc is added or not :+1:
